src/Carbon/Traits/Serialization.php

```diff
@@ -127,9 +128,14 @@ public function __sleep()
      */
     public function __wakeup()
     {
-        if (get_parent_class() && method_exists(parent::class, '__wakeup')) {
+        // FatalError occurs when calling __wakeup method in PHP 7.4 or later.
+        // @codeCoverageIgnoreStart
+        if (version_compare(PHP_VERSION, '7.4.0-dev', '>=')) {
+            parent::__construct($this->{$this->dumpProperties[0]}, $this->timezone !== null ? new DateTimeZone($this->timezone) : null);
```

Comment:
As per my tests when running your code, at this point `$this->date` is null:
```php
var_dump($this->{$this->dumpProperties[0]}, $this->date); // null null
```
And so we actually create a instance with the current datetime instead of getting a copy of the original object.

So unless I'm missing something, this does not fix #2299